scripts/tests/py/metadata.py

```diff
@@ -0,0 +1,225 @@
+        """
+        Reads the YAML file and Constructs the environment object
+
+        Parameters:
+        
+        env_yaml_file_path:
+          Path to the environment file that contains the YAML configuration.
+        """
+        with open(env_yaml_file_path) as stream:
+            self.env=yaml.safe_load(stream)
+
+    def __resolve_env_vals__(self, metadata_dict: Dict[str, Union[str,bool]]) -> None:
+        """
+        Resolves the argument defined in the test script to environment values.
+        For example, if a test script defines "all_clusters" as the value for app
+        name, we will check the environment configuration to see what raw value is
+        assocaited with the "all_cluster" variable and set the value for "app" option
+        to this raw value.
+
+        Parameter:
+        
+        metadata_dict:
+          Dictionary where each key represent a particular argument and its value represent
+          the value for that argument defined in the test script.
+        """
+
+        for run_arg,run_arg_val in metadata_dict.items():
+
+            if not type(run_arg_val)==str or run_arg=="run":
+                metadata_dict[run_arg]=run_arg_val
+                continue
+            
+            if run_arg_val is None:
+                continue
+
+            sub_args = run_arg_val.split('/')
+            
+            if len(sub_args) not in [1,2]:
+                err = """The argument is not in the correct format. 
+                The argument must follow the format of arg1 or arg1/arg2. 
+                For example, arg1 represents the argument type and optionally arg2 
+                represents a specific variable defined in the environment file whose
+                value should be used as the argument value. If arg2 is not specified,
+                we will just use the first value associated with arg1 in the environment file."""
+                raise Exception(err)
+                        
+            if len(sub_args)==1:
+                run_arg_val=self.env.get(sub_args[0])
+                
+            elif len(sub_args)==2:
+                run_arg_val=self.env.get(sub_args[0]).get(sub_args[1])
+
+            # if a argument has been specified in the comment header
+            # but can't be found in the env file, consider it to be
+            # boolean value.
+            if run_arg_val is None:
+                run_arg_val = True
+
+            metadata_dict[run_arg] = run_arg_val
+              
+            
+    def __read_args__(self,run_args_lines: List[str],metadata_dict: Dict[str, str]) -> None:
+        """
+        Parses a list of lines and extracts argument
+        values from it.
+
+        Parameters:
+        
+        run_args_lines:
+          Line in test script header that contains run argumment definition
+
+        metadata_dict:
+          Dictionary where the extracted arguments will be stored.
+          This represents the side effect of this function.
```

Comment:
why do we pass in these dictionaries? Why is the function not `__read_args__(self, lines) -> Dict[str,str]`

The format of the dictionary should be described: some example inputs and dictionary content. Right now I do not know what "run_args_lines` look like (there is a impl comment that says they should be name or name/val) or what the dictionary contains. What is the behaviour on duplicate lines?